go/pkg/cas/upload.go

```diff
@@ -1069,9 +1069,9 @@ func (u *uploader) streamFromReader(ctx context.Context, r io.Reader, digest *re
 		instanceSegment = ""
 	}
 	if compressed {
-		req.ResourceName = fmt.Sprintf("%suploads/%s/compressed-blobs/zstd/%s/%d", instanceSegment, uuid.New(), digest.Hash, digest.SizeBytes)
+		req.ResourceName = fmt.Sprintf("%suploads/%s/compressed-blobs/zstd/%s/%d", instanceSegment, uuid.New().String(), digest.Hash, digest.SizeBytes)
```

Comment:
I didn't end up making a convenience wrapper, since I converted a few of these over to actual error handling instead. An empty string here would/could have caused different failures later on. In the actual optional cases I did add logging.